File: 04-Sorting/01-MaxProductOfThree/ruby/solution.rb
def solution(a)
    a.sort!
    # sort the array and then, due to negatives present in the arrays, we must check the extremes: (-) * (-) * (+) = (+)
    [a[0]*a[1]*a[2], a[0]*a[1]*a[-1], a[0]*a[-1]*a[-2], a[-1]*a[-2]*a[-3]].max
end

File: 04-Sorting/01-MaxProductOfThree/javascript/solution.js
function solution(A) {
    A=A.sort(function(a,b) { return a-b; });
    n=A.length;
    return Math.max(A[0]*A[1]*A[2],A[0]*A[1]*A[n-1],A[0]*A[n-1]*A[n-2],A[n-1]*A[n-2]*A[n-3]);
}